crates/slumber_core/src/template/render.rs

```diff
@@ -724,6 +735,50 @@ impl<'a> ChainTemplateSource<'a> {
         });
         rx.await.map_err(|_| ChainError::PromptNoResponse)
     }
+
+    async fn render_select(
+        &self,
+        context: &'a TemplateContext,
+        stack: &mut RenderKeyStack<'a>,
+        message: Option<&'a Template>,
+        options: &'a [Template],
+    ) -> Result<String, ChainError> {
+        let (tx, rx) = oneshot::channel();
+        let message = if let Some(template) = message {
+            template
+                .render_chain_config("message", context, stack)
+                .await?
+        } else {
+            self.chain_id.to_string()
+        };
+
+        let options = future::try_join_all(options.iter().enumerate().map(
+            |(i, template)| {
+                // Fork the local state, one copy for each new branch
+                let mut stack = stack.clone();
+                async move {
+                    template
+                        .render_chain_config(
+                            format!("options[{i}]"),
+                            context,
+                            &mut stack,
+                        )
+                        .await
+                }
+            },
+        )).await?;
+
+        context.prompter.select(Select {
+            message,
+            options: options.clone(),
+            channel: tx.into(),
+        });
+        let res = rx.await.map_err(|_| ChainError::PromptNoResponse)?;
```

Comment:
I'm fine reusing this error variant, but could you update the message on it in `error.rs` to say `No response from prompt/select`?